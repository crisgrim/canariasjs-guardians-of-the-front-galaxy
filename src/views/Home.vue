<template>
  <div class="home">
    <h1>My <strong>Videogames</strong> List</h1>
    <GamesList :games="games" />
  </div>
</template>

<script>
import GamesList from '@/components/GamesList.vue';

export default {
  name: 'Home',
  components: {
    GamesList,
  },
  data() {
    return {
      games: [
        {
          id: 0,
          title: 'God of war',
          year: 2018,
          publish: new Date(2018, 11, 1),
          started: false,
          ended: false,
          percentage: 0,
          estimated: 20,
          score: 8,
          hype: 10,
          categories: ['Adventure', 'Vikings', 'One Player'],
          image: 'https://static.promodescuentos.com/pepperpdimages/threads/thread_large/default/251090_1.jpg',
        },
        {
          id: 1,
          title: 'The Witcher 3',
          year: 2015,
          publish: new Date(2018, 11, 1),
          started: true,
          ended: false,
          percentage: 0,
          estimated: 150,
          score: 9,
          hype: 10,
          categories: ['Open World', 'Rol', 'One Player'],
          image: 'https://lyricstranslate.com/files/styles/large/public/DVZUGpO.jpg',
        },
        {
          id: 2,
          title: 'Hellblade',
          year: 2017,
          publish: new Date(2018, 11, 1),
          started: false,
          ended: false,
          percentage: 0,
          estimated: 20,
          score: 7,
          hype: 7,
          categories: ['Adventure', 'One Player'],
          image: 'https://i11c.3djuegos.com/juegos/10125/proyecto_ninja_theory__nombre_temporal_/fotos/analisis/proyecto_ninja_theory__nombre_temporal_-4006254.jpg',
        },
        {
          id: 3,
          title: 'Skyrim',
          year: 2011,
          publish: new Date(2018, 11, 1),
          started: true,
          ended: false,
          percentage: 0,
          estimated: 100,
          score: 8,
          hype: 10,
          categories: ['Rol', 'One Player'],
          image: 'https://www.clavecd.es/wp-content/uploads/buy-the-elder-scrolls-5-skyrim-cd-key-pc-download-img1.jpg',
        },
      ],
    }
  }
}
</script>
