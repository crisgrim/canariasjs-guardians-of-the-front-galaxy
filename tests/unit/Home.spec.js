import { shallowMount } from '@vue/test-utils';
import Home from '../../src/views/Home.vue';

describe('Home.spec.js', () => {
  let cmp;

  beforeEach(() => {
    cmp = shallowMount(Home);
    cmp.setData({
      games: [
        {
          id: 0,
          title: 'God of war',
          year: 2018,
          publish: new Date(2018, 11, 1),
          started: false,
          ended: false,
          percentage: 0,
          estimated: 20,
          score: 8,
          hype: 10,
          categories: ['Adventure', 'Vikings', 'One Player'],
          image: 'https://static.promodescuentos.com/pepperpdimages/threads/thread_large/default/251090_1.jpg',
        },
      ],
    });
  });

  it('equals games to array with and element', () => {
    expect(cmp.vm.games).toEqual([{
      id: 0,
      title: 'God of war',
      year: 2018,
      publish: new Date(2018, 11, 1),
      started: false,
      ended: false,
      percentage: 0,
      estimated: 20,
      score: 8,
      hype: 10,
      categories: ['Adventure', 'Vikings', 'One Player'],
      image: 'https://static.promodescuentos.com/pepperpdimages/threads/thread_large/default/251090_1.jpg',
    }]);
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot();
  });
});
