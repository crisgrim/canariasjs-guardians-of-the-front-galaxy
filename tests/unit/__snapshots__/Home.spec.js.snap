// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.spec.js has the expected html structure 1`] = `
<div
  class="home"
>
  <h1>
    My 
    <strong>
      Videogames
    </strong>
     List
  </h1>
   
  <gameslist-stub
    games="[object Object]"
  />
</div>
`;
